src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTestTransformer.java

```diff
@@ -59,20 +76,158 @@ public RubyGapicSurfaceTestTransformer(GapicCodePathMapper rubyPathMapper) {
 
   @Override
   public List<String> getTemplateFileNames() {
-    return ImmutableList.of(SMOKE_TEST_TEMPLATE_FILE);
+    return ImmutableList.of(SMOKE_TEST_TEMPLATE_FILE, UNIT_TEST_TEMPLATE_FILE);
   }
 
   @Override
   public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
-    List<ViewModel> views = new ArrayList<>();
+    ImmutableList.Builder<ViewModel> views = ImmutableList.builder();
+    views.add(createUnitTestView(model, apiConfig));
     for (Interface service : new InterfaceView().getElementIterable(model)) {
       SurfaceTransformerContext context = createContext(service, apiConfig);
       if (context.getInterfaceConfig().getSmokeTestConfig() != null) {
         views.add(createSmokeTestClassView(context));
       }
     }
 
-    return views;
+    return views.build();
+  }
+
+  ///////////////////////////////////// Unit Test ///////////////////////////////////////
+
+  private MockCombinedView createUnitTestView(Model model, ApiConfig apiConfig) {
+    SurfaceNamer namer = new RubySurfaceNamer(apiConfig.getPackageName());
+    ImportSectionView importSection = createTestImportSection(model, apiConfig);
```

Comment:
Rather than duplicate import section knowledge, create a new method in `RubyImportSectionTransformer` and call it here.